shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/type/ShardingRouteEngineFactory.java

```diff
@@ -94,6 +96,15 @@ public static ShardingRouteEngine newInstance(final ShardingRule shardingRule, f
         return getShardingRoutingEngine(shardingRule, sqlStatementContext, shardingConditions, tableNames, props);
     }
     
+    private static Collection<String> getTableNames(final SQLStatementContext sqlStatementContext) {
+        Collection<String> tableNames = new LinkedList<>(sqlStatementContext.getTablesContext().getTableNames());
+        if (sqlStatementContext instanceof InsertStatementContext && null != ((InsertStatementContext) sqlStatementContext).getInsertSelectContext()) {
```

Comment:
`GetTableNames()` is supposed to be a function of `InsertStatementContext`. How abou moving these process to `InsertStatementContext` ?

Reply:
@tristaZero This change was originally for verifying the binding table, maybe revert it is a better choice. The binding table is more suitable for performing verification in ShardingInsertStatementValidator.